tests/benchdnn/graph/flex_rewrite.cpp

```diff
@@ -26,6 +26,166 @@
 
 namespace graph {
 
+void flex_rewrite::rewrite_linked_shape_and_attr(deserialized_graph &dgraph) {
+    for (auto &aop : dgraph.ops_) {
+        if (aop.kind_ == "DynamicDequantize") {
+            auto &attr = aop.attrs_;
+            if (attr.find("qtype") == attr.end()
+                    || attr["qtype"].str_value_ != "per_group")
+                continue;
+            if (attr.find("group_shape") == attr.end()) {
+                BENCHDNN_PRINT(0,
+                        "Error: missed `group-shape` attribute for "
+                        "per-group quantization for op with id=\'%zu\'\n",
+                        aop.id_);
+                SAFE_V(FAIL);
+            }
+
+            bool input_shape_rewrite = std::any_of(aop.in_lts_.begin(),
+                    aop.in_lts_.end(), [&](const deserialized_lt &in_lt) {
+                        return in_shapes_.count(in_lt.id_)
+                                && in_shapes_[in_lt.id_] != "default";
+                    });
+            bool group_shape_rewrite = op_attrs_.count(aop.id_)
+                    && parse_attrs(op_attrs_.at(aop.id_)).count("group_shape");
+
```

Comment:
I see some checks are duplicated, I guess reorganizing code a little bit can help eliminate those duplicates (pseudo-code):
```
if (!input_shape_rewrite && !group_shape_rewrite) continue;

if (input_shape_rewrite) {
    checks_for_src_and_scale
}
if (group_shape_rewrite) {
    checks_for_src_and_group
}
```

And then bodies of action will just do what they need to do.